api/nodes/views.py

```diff
@@ -92,17 +98,60 @@ class NodeDetail(generics.RetrieveUpdateAPIView, NodeMixin):
         ContributorOrPublic,
         ReadOnlyIfRegistration,
     )
-    serializer_class = NodeSerializer
 
-    # overrides RetrieveUpdateAPIView
+    def get_serializer_class(self):
+        if self.request.method == 'DELETE':
+            serializer_class = NodeDeleteSerializer
+            return serializer_class
+        serializer_class = NodeSerializer
+        return serializer_class
+
+    # overrides RetrieveUpdateDestroyAPIView
     def get_object(self):
         return self.get_node()
 
-    # overrides RetrieveUpdateAPIView
+    # overrides RetrieveUpdateDestroyAPIView
     def get_serializer_context(self):
         # Serializer needs the request in order to make an update to privacy
         return {'request': self.request}
 
+    def delete(self, request, node_id):
+        user = self.request.user
+        node = self.get_node()
+        token = node_token_creator(node._id, user._id)
+        url = absolute_reverse('nodes:node-delete-confirm', kwargs={'node_id': node._id, 'token': token})
+        delete_warning = BEFORE_DELETE_NODE.format(_(node.title))
```

Comment:
There's no need to call `ugettext` on node titles, as that is a user input value.